### PR TITLE
Add Octopad+ Keyboard

### PR DESCRIPTION
**Pull request by DeskDaily**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/nightly_boards/octopadplus/info.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    "url": "",
+    "maintainer": "DeskDaily",
+    "usb": {
+    	"vid": "0xD812",
+    	"pid": "0x0014",
+    	"device_version": "0.0.1"
+	},
+	    "rgblight": {
+        "animations": {
+            "all": true
```
Comment (drashna) [inline keyboards/nightly_boards/octopadplus/info.json:13]: Could you explicitly define which animations are enabled? 
```suggestion
            "alternating": true,
            "breathing": true,
            "christmas": true,
            "knight": true,
            "rainbow_mood": true,
            "rainbow_swirl": true,
            "rgb_test": true,
            "snake": true,
            "static_gradient": true,
            "twinkle": true
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/nightly_boards/octopadplus/info.json (hunk 1 above)
Comment (DeskDaily) [reply keyboards/nightly_boards/octopadplus/info.json:13]: Will do thanks

**Review by drashna — APPROVED** [review]

File: keyboards/nightly_boards/octopadplus/keymaps/default/rules.mk [hunk 2]
```diff
@@ -0,0 +1,3 @@
+VIA_ENABLE = yes
```
Comment (zvecr) [inline keyboards/nightly_boards/octopadplus/keymaps/default/rules.mk:1]:
```suggestion
```

File: keyboards/nightly_boards/octopadplus/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
+    [0] =   { ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_BRID, KC_BRIU) },
+    [1] =   { ENCODER_CCW_CW(KC_TRNS, KC_TRNS), ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [2] =   { ENCODER_CCW_CW(KC_TRNS, KC_TRNS), ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [3] =   { ENCODER_CCW_CW(KC_TRNS, KC_TRNS), ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [3] =   { ENCODER_CCW_CW(KC_TRNS, KC_TRNS), ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
```
Comment (zvecr) [inline keyboards/nightly_boards/octopadplus/keymaps/default/keymap.c:53]:
```suggestion
    [3] =   { ENCODER_CCW_CW(KC_TRNS, KC_TRNS), ENCODER_CCW_CW(KC_TRNS, KC_TRNS) }
```

File: keyboards/nightly_boards/octopadplus/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
+    [0] =   { ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_BRID, KC_BRIU) },
+    [1] =   { ENCODER_CCW_CW(KC_TRNS, KC_TRNS), ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [2] =   { ENCODER_CCW_CW(KC_TRNS, KC_TRNS), ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [3] =   { ENCODER_CCW_CW(KC_TRNS, KC_TRNS), ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [3] =   { ENCODER_CCW_CW(KC_TRNS, KC_TRNS), ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
```
Comment (zvecr) [inline keyboards/nightly_boards/octopadplus/keymaps/via/keymap.c:53]:
```suggestion
    [3] =   { ENCODER_CCW_CW(KC_TRNS, KC_TRNS), ENCODER_CCW_CW(KC_TRNS, KC_TRNS) }
```

File: keyboards/nightly_boards/octopadplus/keymaps/via/rules.mk [hunk 5]
```diff
@@ -0,0 +1,3 @@
+VIA_ENABLE = yes
+LTO_ENABLE = yes
```
Comment (zvecr) [inline keyboards/nightly_boards/octopadplus/keymaps/via/rules.mk:2]:
```suggestion
```

File: keyboards/nightly_boards/octopadplus/keymaps/default/rules.mk [hunk 6]
```diff
@@ -0,0 +1,3 @@
...
+LTO_ENABLE = yes
```
Comment (zvecr) [inline keyboards/nightly_boards/octopadplus/keymaps/default/rules.mk:2]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/nightly_boards/octopadplus/keymaps/default/rules.mk (hunk 6 above)
Comment (DeskDaily) [reply keyboards/nightly_boards/octopadplus/keymaps/default/rules.mk:2]: Thanks @zvecr. I already made the changes

File: keyboards/nightly_boards/octopadplus/rules.mk [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes        # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes         # Mouse keys
+EXTRAKEY_ENABLE = yes         # Audio control and System control
+CONSOLE_ENABLE = no           # Console for debug
+COMMAND_ENABLE = no           # Commands for debug and configuration
+LTO_ENABLE = yes              # Link Time Optimization, makes the firmware smaller but some features may not work
+NKRO_ENABLE = yes             # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/nightly_boards/octopadplus/rules.mk:16]:
```suggestion
NKRO_ENABLE = yes             # Enable N-Key Rollover
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/nightly_boards/octopadplus/readme.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
+# DeskDaily Octopad+
+
+![Octopad+](https://i.imgur.com/Sa83N0G.jpg)
```
Comment (drashna) [inline keyboards/nightly_boards/octopadplus/readme.md:3]:
```suggestion
![Octopad+](https://i.imgur.com/Sa83N0Gh.jpg)
```

**Review by fauxpark — APPROVED** [review]

**Pull request merged**